eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,49 +1,49 @@
+  [string]$DevOpsFeed = "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json", # DevOp tool feeds.
+  [string]$CodeOwnerFileLocation = "$env:SYSTEM_DEFAULTWORKINGDIRECTORY/.github/CODEOWNERS", # The place we fetch CODEOWNERS file
+  [string]$VsoVariable = "" # Option of write code owners into devop variable
 )
-$target = $TargetDirectory.ToLower().Trim("/")
-$codeOwnersLocation = Join-Path $RootDirectory -ChildPath ".github/CODEOWNERS"
-$ownedFolders = @{}
 
-if (!(Test-Path $codeOwnersLocation)) {
-  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
-  exit 1
-}
-
-$codeOwnersContent = Get-Content $codeOwnersLocation
 
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+# Check if the retrieve-codeowners tool exsit or not.
+function InstallRetrieveCodeOwnersTool() {
+  if (Get-Command "$ToolPath/retrieve-code-owners") {
+    return
   }
+  dotnet tool install --tool-path $toolPath --add-source "$DevOpsFeed" --version "$ToolVersion" "Azure.Sdk.Tools.RetrieveCodeOwners"
 }
 
-$results = $ownedFolders[$target]
+# Get the json property from tool command.
+function GetCodeOwnersEntryFromCommand() {
+  return & "$ToolPath/retrieve-code-owners" --target-directory "$CodeOwnerPathExpression" --code-owner-file-path "$CodeOwnerFileLocation" 
+}
 
-if ($results) {
-  Write-Host "Found a folder $results to match $target"
+function GetCodeOwners([string]$codeOwnersString) 
+{ 
+  $codeOwnersJson = $codeOwnersString | ConvertFrom-Json
+  if (!$codeOwnersJson) {
+    Write-Host "No code owners returned from the path: $CodeOwnerPathExpression"
+    return ""
+  }
   
+  $codeOwners = $codeOwnersJson.Owners -join ","
+  Write-Host "Code owners are $codeOwners"
   if ($VsoVariable) {
-    $alreadyPresent = [System.Environment]::GetEnvironmentVariable($VsoVariable)
-
-    if ($alreadyPresent) { 
-      $results += ",$alreadyPresent"
-    }
-    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$results"
+    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$codeOwners"
   }
-
-  return $results
+  return $codeOwners
 }
-else {
-  Write-Host "Unable to match path $target in CODEOWNERS file located at $codeOwnersLocation."
-  Write-Host ($ownedFolders | ConvertTo-Json)
+
+InstallRetrieveCodeOwnersTool
+
+$codeOwnerToolOutput = GetCodeOwnersEntryFromCommand
+# Failed at the command of fetching code owners.
+if ($LASTEXITCODE -ne 0) {
+  Write-Host "Not able to fetch the result from command."
+  Write-Host $codeOwnerToolOutput
```

Comment:
This `Write-Host` is a noop and unnecessary. The stderr stream will already be printed out to the terminal when you call the tool, and result variable will only contain what was sent to stdout.

Reply:
Yes, exactly.